parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -137,11 +141,20 @@ public void write(int repetitionLevel, T value) {
     public void setColumnStore(ColumnWriteStore columnStore) {
       this.column.setColumnStore(columnStore);
     }
+
+    @Override
+    public Stream<FieldMetrics> metrics() {
+      if (id != null) {
+        return Stream.of(new FieldMetrics(id.intValue(), 0L, 0L, 0L, null, null));
```

Comment:
We may potentially use `ColumnDescriptor desc` for determine id, but felt that explicitly passing `id` could provide better guarantee. Suggestions are welcomed.

Reply:
Yeah it's a messy change, it would be great if we can somehow not pass in the `id`

Reply:
I think this should be `Stream.empty` unless it is overridden for float and double.

Reply:
I originally implemented it as `Stream.empty` and later added these since I wanted to match the set of ids being reported by NaN counter with the set from value/null counter from `writer.metrics`. But I guess since NaN will never be non-0 for non-float/double fields, and ORC reports a different set of ids than parquet anyway, there's no reason that the sets have to match. I'll update `metrics()` to only create metric object for float and double writers.